From aff5bbf91746bd813126661b8bc76c26401382f7 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 29 Nov 2024 20:37:55 +0100
Subject: [PATCH] Mouse3DController: Add SpaceMouse Wireless BT support

Update the device list to match the upstream spacenavd one.
---
 src/slic3r/GUI/Mouse3DController.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/slic3r/GUI/Mouse3DController.cpp b/src/slic3r/GUI/Mouse3DController.cpp
index 84590d0f976e..1ff1e79d63e6 100644
--- a/src/slic3r/GUI/Mouse3DController.cpp
+++ b/src/slic3r/GUI/Mouse3DController.cpp
@@ -27,7 +27,7 @@ static const std::vector<int> _3DCONNEXION_VENDORS =
     0x256F   // 3DCONNECTION = 9583 // 3Dconnexion
 };
 
-// See: https://github.com/FreeSpacenav/spacenavd/blob/a9eccf34e7cac969ee399f625aef827f4f4aaec6/src/dev.c#L202
+// See: https://github.com/FreeSpacenav/spacenavd/blob/39856625a6de1e8c4b57c5938e1bf29d13cf1a9f/src/dev.c#L63
 static const std::vector<int> _3DCONNEXION_DEVICES =
 {
     0xc603,	/* 50691 spacemouse plus XT */
@@ -41,6 +41,7 @@ static const std::vector<int> _3DCONNEXION_DEVICES =
     0xc628,	/* 50728 space navigator for notebooks*/
     0xc629,	/* 50729 space pilot pro*/
     0xc62b,	/* 50731 space mouse pro*/
+    0xc640,	/* 50752 nulooq */
     0xc62e,	/* 50734 spacemouse wireless (USB cable) *TESTED* */
     0xc62f,	/* 50735 spacemouse wireless receiver */
     0xc631,	/* 50737 spacemouse pro wireless *TESTED* */
@@ -48,7 +49,7 @@ static const std::vector<int> _3DCONNEXION_DEVICES =
     0xc633,	/* 50739 spacemouse enterprise */
     0xc635,	/* 50741 spacemouse compact *TESTED* */
     0xc636,	/* 50742 spacemouse module */
-    0xc640,	/* 50752 nulooq */
+    0xc63a,	/* 60060 spacemouse wireless (Bluetooth) */
     0xc652, /* 50770 3Dconnexion universal receiver *TESTED* */
 };
 
-- 
2.47.1

